Compilation fails with G++

```
In file included from src/parser/statements/../expressions/integer_constant.hpp:5,
                 from src/parser/statements/../parser.hpp:11,
                 from src/parser/statements/while_loop.hpp:5,
                 from src/parser/statements/while_loop.cpp:1:
src/parser/statements/../expressions/../common.hpp:11:15: error: declaration of 'koala::type* koala::variable_info::type' changes meaning of 'type' [-Wchanges-meaning]
   11 |         type* type;
      |               ^~~~
src/parser/statements/../expressions/../common.hpp:11:9: note: used here to mean 'class koala::type'
   11 |         type* type;
      |         ^~~~
In file included from src/parser/statements/../parser.hpp:9:
src/parser/statements/../type_system.hpp:21:11: note: declared here
   21 |     class type {
      |           ^~~~
make: *** [bin/koala] Error 1
```
This can be fixed by just adding 'class' or 'struct' to the member variable declaration.

--- a/src/parser/common.hpp
+++ b/src/parser/common.hpp
@@ -8,6 +8,6 @@ namespace koala {
     struct variable_info {
         std::string name;
 
-        type* type;
+        class type* type;
     };
 }

